src/Worker/Delivery.php

```diff
@@ -278,7 +278,7 @@ private static function setFailedQueue(string $cmd, array $item)
 	private static function deliverDFRN(string $cmd, array $contact, array $owner, array $items, array $target_item, bool $public_message, bool $top_level, bool $followup, int $server_protocol = null)
 	{
 		// Transmit Diaspora reshares via Diaspora if the Friendica contact support Diaspora
-		if (Diaspora::getReshareDetails($target_item ?? []) && !empty(FContact::getByURL($contact['addr'], false))) {
+		if (Diaspora::getReshareDetails($target_item ?? []) && DI::dsprContact()->existsByUri(new Uri($contact['url']))) {
```

Comment:
What is this `new Uri($contact['url'])` doing here? In the Diaspora protocol the contact is addressed by the handle (`user@domain.tld`) instead of the URI.

Reply:
I know, but `user@domain.tld` is a valid `Uri` (although not a valid URL but that's irrelevant here).

Reply:
I don't understand what this `new Uri($contact['url'])` is performing here. Also why don't we keep searching for the handle here? The handle is expected to be the unique identifier in the Diaspora protocol, the resolvable HTTP address is just for convenience (while in ActivityPub it's the resolvable HTTP address and the handle is just for convenience).

Reply:
> I don't understand what this `new Uri($contact['url'])` is performing here.

Ideally, the `url` property of a `Contact` entity should be a `Uri` object. Since it still is a string off an arbitrary array, we need to convert it to the object before interacting with the DiasporaContact methods.

> Also why don't we keep searching for the handle here? The handle is expected to be the unique identifier in the Diaspora protocol, the resolvable HTTP address is just for convenience (while in ActivityPub it's the resolvable HTTP address and the handle is just for convenience).

Because the `addr(32)` key isn't `UNIQUE` in the `dcontact` table, unlike the `url` via the `uri-id`. It isn't that big of a deal in the context of an `exists` query, but retrieving a single DiasporaContact could have unpredictable results with a non-unique key.